app/code/Magento/CatalogSearch/Model/Search/FilterMapper/TermDropdownStrategy.php

```diff
@@ -25,46 +26,37 @@ class TermDropdownStrategy implements FilterStrategyInterface
      */
     private $aliasResolver;
 
-    /**
-     * @var StoreManagerInterface
-     */
-    private $storeManager;
-
     /**
      * @var EavConfig
      */
     private $eavConfig;
 
     /**
-     * @var ResourceConnection
-     */
-    private $resourceConnection;
-
-    /**
-     * @var ScopeConfigInterface
+     * @var JoinAdderToSelect
      */
-    private $scopeConfig;
+    private $joinAdderToSelect;
 
     /**
      * @param StoreManagerInterface $storeManager
      * @param ResourceConnection $resourceConnection
      * @param EavConfig $eavConfig
      * @param ScopeConfigInterface $scopeConfig
      * @param AliasResolver $aliasResolver
-     * @SuppressWarnings(Magento.TypeDuplication)
+     * @param JoinAdderToSelect $joinAdderToSelect
+     *
+     * @SuppressWarnings(PHPMD.UnusedFormalParameter)
      */
     public function __construct(
         StoreManagerInterface $storeManager,
         ResourceConnection $resourceConnection,
         EavConfig $eavConfig,
         ScopeConfigInterface $scopeConfig,
-        AliasResolver $aliasResolver
+        AliasResolver $aliasResolver,
+        JoinAdderToSelect $joinAdderToSelect = null
```

Comment:
StoreManagerInterface $storeManager,
        EavConfig $eavConfig,
        ScopeConfigInterface $scopeConfig,

Type hinting should be removed

Reply:
Done.